src/main/java/teammates/ui/webapi/Action.java

```diff
@@ -111,6 +113,27 @@ public void checkAccessControl() throws UnauthorizedAccessException {
         checkSpecificAccessControl();
     }
 
+    /**
+     * Gets the user information of the current user.
+     */
+    public Map<String, String> getUserInfo() {
+        Map<String, String> info = new HashMap<>();
+
+        String googleId = userInfo == null ? null : userInfo.getId();
+        String regkey = null;
+        String studentEmail = null;
+        Optional<StudentAttributes> student = getUnregisteredStudent();
```

Comment:
Two important things here:
- Regkey can belong to an instructor. Although instructor is required to register to use the system (this will change in the near future, by the way), there is a brief time period where an instructor operates with only regkey.
- Although this works, this is an unnecessary work duplication (there is a logic layer + DB call here). Much like how `userInfo` has been pre-processed in the action itself and simply retrieved here, `regkey` and `email` (which really is not just "student" email) should be the same.

Reply:
Understood. I will make changes for these.

Reply:
Variables are created to store the user information of those without an account so that there will be no duplication.